src/server/game/Handlers/CalendarHandler.cpp

```diff
@@ -47,196 +47,118 @@ Copied events should probably have a new owner
 #include "GuildMgr.h"
 #include "ArenaTeamMgr.h"
 #include "WorldSession.h"
+#include "CalendarPackets.h"
 
-void WorldSession::HandleCalendarGetCalendar(WorldPacket& /*recvData*/)
+void WorldSession::HandleCalendarGetCalendar(WorldPackets::Calendar::CalendarGetCalendar& /*calendarGetCalendar*/)
 {
     ObjectGuid guid = _player->GetGUID();
-    TC_LOG_DEBUG("network", "CMSG_CALENDAR_GET [%s]", guid.ToString().c_str());
-
     time_t currTime = time(NULL);
 
-    WorldPacket data(SMSG_CALENDAR_SEND_CALENDAR, 1000); // Average size if no instance
+    WorldPackets::Calendar::CalendarSendCalendar packet;
+    packet.ServerNow = currTime;
+    packet.RaidOrigin = 1135753200; // Constant date, unk (28.12.2005 07:00)
+    packet.ServerTime = currTime;
 
-    CalendarInviteStore invites = sCalendarMgr->GetPlayerInvites(guid);
-    data << uint32(invites.size());
-    for (CalendarInviteStore::const_iterator itr = invites.begin(); itr != invites.end(); ++itr)
+    CalendarInviteStore playerInvites = sCalendarMgr->GetPlayerInvites(guid);
+    for (auto const& invite : playerInvites)
     {
-        data << uint64((*itr)->GetEventId());
-        data << uint64((*itr)->GetInviteId());
-        data << uint8((*itr)->GetStatus());
-        data << uint8((*itr)->GetRank());
-
-        if (CalendarEvent* calendarEvent = sCalendarMgr->GetEvent((*itr)->GetEventId()))
-        {
-            data << uint8(calendarEvent->IsGuildEvent());
-            data << calendarEvent->GetCreatorGUID().WriteAsPacked();
-        }
-        else
-        {
-            data << uint8(0);
-            data << (*itr)->GetSenderGUID().WriteAsPacked();
-        }
+        WorldPackets::Calendar::CalendarSendCalendarInviteInfo inviteInfo;
+        inviteInfo.EventID = invite->GetEventId();
+        inviteInfo.InviteID = invite->GetInviteId();
+        inviteInfo.InviterGuid = invite->GetSenderGUID();
+        inviteInfo.InviteType = 1; // Always 1 ?
```

Comment:
Looking at the old code only, this seems to be 1 it's a guild event and 0 when it's not.